File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/abstract_factory/SqlserverUser.java
package jv.com.tencent.cain.design_pattern.abstract_factory;

/**
 * @author : jiangyu
 * @date : 2020/3/20
 * @desc : xxx
 */
public class SqlserverUser implements IUser {

    @Override
    public void insertUser(User user) {
        System.out.println("sqlserver# insert user: " + user.toString());
    }

    @Override
    public User queryUser(int id) {
        return new User("cain", 28);
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/abstract_factory/SqlserverUser.kt
package kt.com.tencent.cain.desigin_pattern.abstract_factory

/**
 * @author : jiangyu
 * @date   : 2020/3/20
 * @desc   : xxx
 */
class SqlserverUser : IUser {

    override fun insertUser(user: User) {
        println("sqlserver# insert user: $user")
    }

    override fun queryUser(id: Int): User = User("cain", 30)
}